PathfinderAPI/Util/XML/ElementInfo.cs

```diff
@@ -28,60 +45,222 @@ public override string ToString()
+        => Content.TryAsBoolean(out result);
+
+    public bool TryContentAsInt(out int result)
+        => Content.TryAsInt(out result);
+
+    public bool TryContentAsFloat(out float result)
+        => Content.TryAsFloat(out result);
+
+    public bool GetContentAsBoolean(bool defaultVal = default)
+        => TryContentAsBoolean(out var result) ? result : defaultVal;
+
+    public int GetContentAsInt(int defaultVal = default)
+        => TryContentAsInt(out var result) ? result : defaultVal;
+
+    public float GetContentAsFloat(float defaultVal = default)
+        => TryContentAsFloat(out var result) ? result : defaultVal;
+
     public bool ContentAsBoolean()
-        => bool.TryParse(Content, out var value)
-            ? value
-            : throw new FormatException($"Value of '{Name}' is not true or false");
+        => Content.AsBoolean(nameof(Content));
 
     public int ContentAsInt()
-        => int.TryParse(Content, out var value)
-            ? value
-            : throw new FormatException($"Value of '{Name}' is not an integer, e.g.: 0, 1, 2");
-        
+        => Content.AsInt(nameof(Content));
+
     public float ContentAsFloat()
-        => float.TryParse(Content, out var value)
-            ? value
-            : throw new FormatException($"Value of '{Name}' is not a float, e.g.: 1.0");
+        => Content.AsFloat(nameof(Content));
+
+    public bool TryAttributeAsBoolean(string attribName, out bool result)
+        => Attributes.TryAsBoolean(attribName, out result);
+
+    public bool TryAttributeAsInt(string attribName, out int result)
+        => Attributes.TryAsInt(attribName, out result);
+
+    public bool TryAttributeAsFloat(string attribName, out float result)
+        => Attributes.TryAsFloat(attribName, out result);
+
+    public bool GetAttributeAsBoolean(string attribName, bool defaultVal = default)
+        => TryAttributeAsBoolean(attribName, out var result) ? result : defaultVal;
+
+    public int GetAttributeAsInt(string attribName, int defaultVal = default)
+        => TryAttributeAsInt(attribName, out var result) ? result : defaultVal;
+
+    public float GetAttributeAsFloat(string attribName, float defaultVal = default)
+        => TryAttributeAsFloat(attribName, out var result) ? result : defaultVal;
+
+    public bool AttributeAsBoolean(string attribName)
+        => Attributes.AsBoolean(attribName, $"{nameof(Attributes)}");
+
+    public int AttributeAsInt(string attribName)
+        => Attributes.AsInt(attribName, $"{nameof(Attributes)}");
+
+    public float AttributeAsFloat(string attribName)
+        => Attributes.AsFloat(attribName, $"{nameof(Attributes)}");
+
+    public bool TryAddChild(ElementInfo info)
+    {
+        if(Children == null) return false;
+        if(info.Parent != null)
+            info.Parent.Children.Remove(info);
+        Children.Add(info);
+        info.Parent = this;
+        return Children.Last() == info;
+    }
+
+    public bool TrySetParent(ElementInfo info)
```

Comment:
I don't think this makes sense to be here, an element doesn't doesn't care about its parent, so why would you set it through the child element

Reply:
I suppose, was doing it for convenience but I suppose a child doesn't need to be aware of its parent's add success